api/v1beta1/moduleimagesconfig_types.go

```diff
@@ -54,7 +54,7 @@ type ModuleImageSpec struct {
 // More info: https://git.k8s.io/community/contributors/devel/sig-architecture/api-conventions.md#spec-and-status
 // +kubebuilder:validation:Required
 type ModuleImagesConfigSpec struct {
-	Images []ModuleImageSpec `json:"images"`
+	Images []ModuleImageSpec `json:"images,omitempty"`
```

Comment:
I would suggest a different approach.

Creating an MIC object with no images in it have no effect and doesn't makes sense. I think we should instead add a condition and only create the MIC if `len(images)>0`.